documentation/release_notes.markdown

```diff
@@ -109,131 +162,150 @@ stable release of PuppetDB, which included an experimental v4 API.
 
-* Add pagination to service query method ([PDB-1071](https://tickets.puppetlabs.com/browse/PDB-1071))
+* We now store report logs ([PDB-1192](https://tickets.puppetlabs.com/browse/PDB-1184))
 
-    Add a `paging_options` argument to the service query method.
+* We have added `count` aggregation capabilities to the query API ([PDB-1181](https://tickets.puppetlabs.com/browse/PDB-1181))
 
-* Make `/pdb/query/v4/catalogs` queryable ([PDB-1028](https://tickets.puppetlabs.com/browse/PDB-1028))
+* Support `extract` as a top level query operator ([PDB-207](https://tickets.puppetlabs.com/browse/PDB-207))
 
-* Add `producer_timestamp` field to factsets and catalogs ([PDB-489](https://tickets.puppetlabs.com/browse/PDB-489))
+    This provides support for requesting only the fields that you need in
+    an API response when querying data.
 
-    On factsets the field is queryable and orderable.
+* Support fallback PuppetDB connections ([PDB-100](https://tickets.puppetlabs.com/browse/PDB-100))
 
-* Add `resource_events` key to the `store report` command ([PDB-1072](https://tickets.puppetlabs.com/browse/PDB-1072))
+    When configuring your connection to PuppetDB, you can now specify a fallback.
+    Instead of the old `server` and `port` config, you can now use a
+    `server\_urls` config that contains the full path to one or more PuppetDB
+    URLs. The old server/port format is still supported.
 
-    API responses can now be easily used for resubmission as a
-    report. This will be helpful in the HA context, where it will reduce
-    the number of requests needed to synchronize reports.
+    This commit also supports a new `server\_url\_timeout` config. If the
+    PuppetDB instance has not responded in the specified number of
+    seconds (defaults to 30) it will fail and roll to the next PuppetDB
+    url (if one is provided).
 
-* Support `extract` as a top level query operator ([PDB-207](https://tickets.puppetlabs.com/browse/PDB-207))
+* Add support for hyphenated classnames, supported by Puppet 2.7.x - 3.7.x. ([PDB-1024](https://tickets.puppetlabs.com/browse/PDB-1024))
+
+* Add a `-v` or `--version` flag to the CLI to get the PuppetDB version. ([PDB-992](https://tickets.puppetlabs.com/browse/PDB-922))
+
+* The reports endpoint now implements the `latest_report?` query, which
+  filters the response to only include reports that are from the
+  latest puppet run for each node.
+  ([PDB-1244](https://tickets.puppetlabs.com/browse/PDB-1244))
 
-    This supports a single column or multiple columns. It has been tested for
-    facts, events, nodes, reports and resources
+* We have now added [edges](./api/query/v4/edges.html) querying capability to the top-level namespace. You can also query edges specific to a particular catalog using the new `edges` suffix on the [catalogs](./api/query/v4/catalogs.html) endpoint. ([PDB-1228](https://tickets.puppetlabs.com/browse/PDB-1228)).
```

Comment:
top-level namespace isn't accurate with the pathing changes. We have added an edges endpoint to the query API.